### PR TITLE
feat: add posthog-js lib and website_growth_compute_price_rising exp

### PR DESCRIPTION
**Pull request by bacher**
Relates to https://linear.app/neondb/project/raise-compute-price-to-dollar024hr-on-public-pricing-page-c25729a09d29/overview
Closes https://linear.app/neondb/issue/GRW-290/experiment-build

This PR adds `posthog-js` https://www.npmjs.com/package/posthog-js library based on https://posthog.com/docs/libraries/next-js documentation.
`PostHogProvider.jsx` and `PostHogPageView.jsx` are copied from documentation.

And as next step I added the first experiment https://us.posthog.com/project/49356/experiments/68916
Feature flag: `website_growth_compute_price_rising`

The diff is quite large but mostly because I had to change indentation in 2 components, so please look at diff with `Hide whitespace` enabled.

Before this PR is merged we have to do 2 thinkgs.

1. Disable injection of posthog client using GTM or it's analog. I see that we embed it somehow:

image

---

2. And somebody have to add these env variables into Vercel's Production configuration:
```
NEXT_PUBLIC_POSTHOG_KEY=phc_yv8ksk9v2FajpXIxVHYj4g59LVeLyo8yN5SaTG1JJty
NEXT_PUBLIC_POSTHOG_HOST=https://us.i.posthog.com
```

### CODE REVIEW

File: src/app/posthog-provider.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+// Provider configured based by the official Posthog documentation:
+// https://posthog.com/docs/libraries/next-js
+
+const PostHogProvider = ({ children }) => {
+  useEffect(() => {
+    const key = process.env.NEXT_PUBLIC_POSTHOG_KEY;
+
+    if (key) {
+      posthog.init(key, {
```
Comment (duskpoet) [inline src/app/posthog-provider.jsx:16]: let's use the same proxy hosts we use in console? 
https://github.com/neondatabase/cloud/blob/main/web/analytics/analytics/provider.tsx#L35

File: src/components/pages/pricing/plans/table/table.jsx [hunk 2]
```diff
@@ -295,6 +336,12 @@ const Table = () => {
                   theme={isHighlightedColumn ? 'primary' : 'gray-15'}
                   to={labelList[key].buttonUrl}
                   tag_name={`Details Table Bottom > ${labelList[key].label}`}
+                  onClick={() => {
+                    const event = labelList[key].analyticsEvent;
+                    if (event.analyticsEvent) {
```
Comment (duskpoet) [inline src/components/pages/pricing/plans/table/table.jsx:341]: I think you have an extra property access here

**Review by duskpoet — CHANGES_REQUESTED** [review]

File: src/components/pages/pricing/plans/table/table.jsx (hunk 2 above)
Comment (bacher) [reply src/components/pages/pricing/plans/table/table.jsx]: Oh, thank you! Website is written in Javascript, with all disadvantages...

Fixed!

File: src/app/posthog-provider.jsx (hunk 1 above)
Comment (bacher) [reply src/app/posthog-provider.jsx:16]: I even don't access to our segment account anymore.

Comment (duskpoet) [reply src/app/posthog-provider.jsx:16]: oh, stupid me, sorry)

**Review by duskpoet — APPROVED** [review]

**Comment by bacher** [comment]
> I've updated prod envs
> 
> Btw, there is some issue with build

@duskpoet 
Looks like I don't have an access. Since you updated envs I suppose you do have access. Can you send me a screenshot of error from Vercel which relates to my build?

image

**Pull request merged**